### PR TITLE
Add foreign keys

### PR DESCRIPTION
**Pull request by j16r**
Add foreign keys to the suggested models.

Note that in gorm creation/saving has a difference: insert vs update. This difference becomes more important with foreign keys. This change has been made more explicit in the code.

Other changes of note:

  1. Service Agreement and JobSpec get made in one go
  2. Pragma to enable foreign keys, may be ignored on postgres
  3. Fixture methods created two JobSpecs with the same ID, this has been fixed.

### CODE REVIEW

File: store/orm/orm.go [hunk 1]
```diff
@@ -195,7 +197,7 @@ func (orm *ORM) SaveJob(job *models.JobSpec) error {
 	for i := range job.Initiators {
 		job.Initiators[i].JobSpecID = job.ID
 	}
-	return orm.DB.Save(job).Error
+	return orm.DB.Create(job).Error
```
Comment (dimroc) [inline store/orm/orm.go]: Maybe we should go ahead and rename the method `CreateJob` to reflect this. And it's a great opportunity to use `:GoRename`

Comment (j16r) [reply store/orm/orm.go]: That was fun ;)

File: internal/cltest/cltest.go [hunk 2]
```diff
@@ -613,16 +613,25 @@ func CreateJobRunViaWeb(t *testing.T, app *TestApplication, j models.JobSpec, bo
...
+	job.Tasks[0].Params = JSONFromString(fmt.Sprintf(`{"url":"%v"}`, url))
+	return CreateJobSpecViaWeb(t, app, job)
 }
 
 // CreateMockAssignmentViaWeb creates a JobSpec with the given MockServer Url
 func CreateMockAssignmentViaWeb(t *testing.T, app *TestApplication, url string) models.JobSpec {
 	j := FixtureCreateJobWithAssignmentViaWeb(t, app, "../internal/fixtures/web/v1_format_job.json")
 	j.Tasks[0].Params = JSONFromString(fmt.Sprintf(`{"url":"%v"}`, url))
-	return CreateJobSpecViaWeb(t, app, j)
+	return j
```
Comment (dimroc) [inline internal/cltest/cltest.go]: Perhaps it's not an issue now, but having a job spec saved different to what's returned in memory will bite us. Can we fix this like we fixed the one above (`CreateHelloWorldJobViaWeb`)?

File: internal/cltest/cltest.go [hunk 3]
```diff
@@ -613,16 +613,25 @@ func CreateJobRunViaWeb(t *testing.T, app *TestApplication, j models.JobSpec, bo
...
+	buffer, err := ioutil.ReadFile("../internal/fixtures/web/hello_world_job.json")
+	if err != nil {
+		assert.FailNowf(t, "Unable to read fixture", err.Error())
+	}
+
+	var job models.JobSpec
+	err = json.Unmarshal(buffer, &job)
+	require.NoError(t, err)
+
+	job.ID = utils.NewBytes32ID()
```
Comment (dimroc) [inline internal/cltest/cltest.go]: I think we can ✂️ cut this line. The `ID` gets populated server side when the spec is created.

Comment (j16r) [reply internal/cltest/cltest.go]: It does not work without this, digging into why...

Comment (j16r) [reply internal/cltest/cltest.go]: Fixed.

File: internal/cltest/cltest.go (hunk 2 above)
Comment (j16r) [reply internal/cltest/cltest.go]: Yeah I'm cleaning this one up, but it's a bit more complicated than it initially seems!

File: web/service_agreements_controller.go [hunk 4]
```diff
@@ -40,10 +40,7 @@ func (sac *ServiceAgreementsController) Create(c *gin.Context) {
 		} else if err = services.ValidateServiceAgreement(sa, sac.App.GetStore()); err != nil {
 			publicError(c, 422, err)
 			return
-		} else if err = sac.App.GetStore().SaveServiceAgreement(&sa); err != nil {
-			_ = c.AbortWithError(500, err)
-			return
-		} else if err = sac.App.AddJob(sa.JobSpec); err != nil {
+		} else if err = sac.App.AddServiceAgreement(&sa); err != nil {
```
Comment (dimroc) [inline web/service_agreements_controller.go]: 👍

**Pull request merged**